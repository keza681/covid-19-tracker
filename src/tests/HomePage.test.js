/* eslint-disable camelcase */
import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from '../redux/store';
import Countries from '../components/HomePage';

describe('Homepage test', () => {
  it('Should display item from api', () => {
    const app = renderer
      .create(
        <Provider store={configureStore}>
          <Countries />
        </Provider>,
      )
      .toJSON();
    expect(app).toMatchSnapshot();
  });

  it('Should render detail page by rendering', () => {
    const path = renderer
      .create(
        <Provider store={configureStore}>
          <Countries />
        </Provider>,
      )
      .toJSON();
    expect(path).toMatchSnapshot();
  });
});
