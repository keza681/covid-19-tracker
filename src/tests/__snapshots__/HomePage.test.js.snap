// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page test countries container test 1`] = `
<div>
  <header
    className="cover"
  >
    <div
      className="title"
    >
      <div
        className="logo"
      >
        <div>
          <h1>
            Covid Cases
          </h1>
        </div>
      </div>
    </div>
  </header>
  <div
    className="search"
  >
    <div
      className="myDate"
    >
      <h2>
         Stats by country
      </h2>
      <h3
        className="date"
      >
        2022-04-28
      </h3>
    </div>
    <input
      className="inputSearch"
      id="search"
      name="search"
      onChange={[Function]}
      placeholder="search here"
      type="text"
      value=""
    />
  </div>
  <ul
    className="myCounty"
  />
</div>
`;

exports[`Home page test country cards test 1`] = `
<div
  className="card"
>
  <a
    className=""
    href="/Iran"
    onClick={[Function]}
  >
    <li
      className="countryCard"
    >
      <div
        className="mapBox"
      >
        <img
          alt="covid img"
          className="map"
          src="https://cdn.pixabay.com/photo/2020/04/29/07/54/coronavirus-5107715_960_720.png"
        />
      </div>
      <p>
        Iran
      </p>
      <h4 />
    </li>
  </a>
</div>
`;

exports[`Homepage test Should also render diffetent part of the app by routing 1`] = `
<div>
  <header
    className="cover"
  >
    <div
      className="title"
    >
      <div
        className="logo"
      >
        <div>
          <h1>
            Covid Cases
          </h1>
        </div>
      </div>
    </div>
  </header>
  <div
    className="search"
  >
    <div
      className="myDate"
    >
      <h2>
         Stats by country
      </h2>
      <h3
        className="date"
      >
        2022-04-28
      </h3>
    </div>
    <input
      className="inputSearch"
      id="search"
      name="search"
      onChange={[Function]}
      placeholder="search here"
      type="text"
      value=""
    />
  </div>
  <ul
    className="myCounty"
  />
</div>
`;

exports[`Homepage test Should display item from api 1`] = `
<div>
  <header
    className="cover"
  >
    <div
      className="title"
    >
      <div
        className="logo"
      >
        <div>
          <h1>
            Covid Cases
          </h1>
        </div>
      </div>
    </div>
  </header>
  <div
    className="search"
  >
    <div
      className="myDate"
    >
      <h2>
         Stats by country
      </h2>
      <h3
        className="date"
      >
        2022-04-28
      </h3>
    </div>
    <input
      className="inputSearch"
      id="search"
      name="search"
      onChange={[Function]}
      placeholder="search here"
      type="text"
      value=""
    />
  </div>
  <ul
    className="myCounty"
  />
</div>
`;

exports[`Homepage test Should render detail page by rendering 1`] = `
<div>
  <header
    className="cover"
  >
    <div
      className="title"
    >
      <div
        className="logo"
      >
        <div>
          <h1>
            Covid Cases
          </h1>
        </div>
      </div>
    </div>
  </header>
  <div
    className="search"
  >
    <div
      className="myDate"
    >
      <h2>
         Stats by country
      </h2>
      <h3
        className="date"
      >
        2022-04-28
      </h3>
    </div>
    <input
      className="inputSearch"
      id="search"
      name="search"
      onChange={[Function]}
      placeholder="search here"
      type="text"
      value=""
    />
  </div>
  <ul
    className="myCounty"
  />
</div>
`;

exports[`The test of welcome page component Should also render diffetent part of the app by routing 1`] = `
<div>
  <header
    className="cover"
  >
    <div
      className="title"
    >
      <div
        className="logo"
      >
        <div>
          <h1>
            Covid Cases
          </h1>
        </div>
      </div>
    </div>
  </header>
  <div
    className="search"
  >
    <div
      className="myDate"
    >
      <h2>
         Stats by country
      </h2>
      <h3
        className="date"
      >
        2022-04-28
      </h3>
    </div>
    <input
      className="inputSearch"
      id="search"
      name="search"
      onChange={[Function]}
      placeholder="search here"
      type="text"
      value=""
    />
  </div>
  <ul
    className="myCounty"
  />
</div>
`;

exports[`The test of welcome page component Should display all the rendered items from api 1`] = `
<div>
  <header
    className="cover"
  >
    <div
      className="title"
    >
      <div
        className="logo"
      >
        <div>
          <h1>
            Covid Cases
          </h1>
        </div>
      </div>
    </div>
  </header>
  <div
    className="search"
  >
    <div
      className="myDate"
    >
      <h2>
         Stats by country
      </h2>
      <h3
        className="date"
      >
        2022-04-28
      </h3>
    </div>
    <input
      className="inputSearch"
      id="search"
      name="search"
      onChange={[Function]}
      placeholder="search here"
      type="text"
      value=""
    />
  </div>
  <ul
    className="myCounty"
  />
</div>
`;
